datafusion-examples/examples/rewrite_expr.rs

```diff
@@ -59,7 +59,7 @@ pub fn main() -> Result<()> {
 
     // then run the optimizer with our custom rule
     let optimizer = Optimizer::with_rules(vec![Arc::new(MyOptimizerRule {})]);
-    let optimized_plan = optimizer.optimize(&analyzed_plan, &config, observe)?;
+    let optimized_plan = optimizer.optimize(analyzed_plan, &config, observe)?;
```

Comment:
This is the only API change exposed to users -- `Optimizer::optimize` takes an owned `LogicalPlan` rather than a `&LogicalPlan`. This is to avoid the need to copy over and over again